### ISSUE TITLE
Is that what you meant 

### ISSUE THREAD

**Issue by LazzaAU** [project-alice-assistant/skill_AliceCore#15]
The feature " is that what you want " needs fine tuning.

**test Case**

first request i purposely requested wrong - " turn on the long light"
second request (and correct request) -- "turn on the awning light"

Alice responds " is this what you meant when you said turn on the long light ?
I replied "yes"
and she went into training

All good that's what i expected and wanted. However .....

In dialogTemplate of "AliceCore" she added "turn on the long light" to the slot type of "Alice/AnswerYesNo" , which in my mind is wrong. ??

she should be
 -  finding the skill that has the correct intent ( in this case HA skill)
 -  then adding "long light" as a **synonym**  for the slot value that the correct intent is using ( in this case HA skill slot "swichNames" value "awning light")

**Comment by Psychokiller1888** [comment]
Yep, never saw that, weird. Of course it should add the utterance to the correct intent

**Comment by LazzaAU** [comment]
In somewhere like slotsAsObjects, would it be possible to add a result which is something like

``` "callingSkill" : "Path"``` ,

 where "Path" would be the path to the skill of the **"recognised intent"**

I was  just looking into ways to correct the above "is that what you meant" , and to be able to add the correct synonym or utterance to the correct dialogTemplate file we would first need to find out which **skill** had the **correct Intent** and i can't see in the current session where that would be possible. I don't feel self.getResource() is not a feesible option in this senario 

Just a thought

**Comment by Psychokiller1888** [comment]
I just pushed a possible fix: https://github.com/project-alice-assistant/ProjectAlice/commit/39e37b6ab56572b0c34eed9f6a1066c6ba711018

**Comment by LazzaAU** [comment]
*transferring comment from that commit to here:*

:smiles: Ok so maybe i was overthinking that part of it a little. Works good. Puts the utterance under the right intent.

however i'd suspect at some stage phase two would be to work out how to add slots to that intent or put it as a synonym ? as the utterance on its own will only trigger the intent not the actions/slots inside the intent ? or am i still over thinking it :)
